python/paddle/fluid/framework.py

```diff
@@ -5305,6 +5460,155 @@ def all_parameters(self):
+            if var.desc.type() == core.VarDesc.VarType.FEED_MINIBATCH or \
+                var.desc.type() == core.VarDesc.VarType.FETCH_LIST or \
+                var.desc.type() == core.VarDesc.VarType.READER:
+                return False
+            return var.persistable
+
+        def is_belong_to_optimizer(var):
+            if not (isinstance(var, Parameter) or var.desc.need_check_feed()):
+                return is_persistable(var)
+            return False
+
+        def condition(var):
+
+            if mode == 'param':
+                return is_parameter(var)
+            elif mode == 'opt':
+                return is_belong_to_optimizer(var)
+            elif mode == 'all':
+                return is_parameter(var) or is_belong_to_optimizer(var)
+            else:
+                raise ValueError(
+                    "`mode` string should be 'param', 'opt' or 'all', but received {}.".
+                    format(mode))
+
+        var_list = filter(condition, self.list_vars())
+
+        state_dict = dict()
+        for var in var_list:
+            var_temp = scope.find_var(var.name)
+            if var_temp is None:
+                raise ValueError(
+                    "Can not find Variable '{}' in the scope. Make sure it is initialized".
+                    format(var_name))
+            state_dict[var.name] = var_temp.get_tensor()
+
+        return state_dict
+
+    def set_state_dict(self, state_dict, scope=None):
+        """
+        Set program parameter from state_dict
+
+        An exception will throw if shape or dtype of the parameters is not match.
+
+        NOTICE: This function MUST called after run start_up_program
+
+        Args:
+            state_dict(dict): the dict store Parameter and optimizer information
+            scope(Scope, optional) : If scope is None, state_dict will be set to global scope 
+                obtained through 'global_scope()'. Otherwise, value will be set to scope.
+                Default: None
+        Returns:
+            None
+
+        Examples:
+            .. code-block:: python
+
+            import paddle
+            import paddle.fluid as fluid
+
+            paddle.enable_static()
+
+            x = fluid.data(name="x", shape=[10, 10], dtype='float32')
+            y = fluid.layers.fc(x, 10)
+            z = fluid.layers.fc(y, 10)
+
+            place = paddle.CPUPlace()
+            exe = fluid.Executor(place)
+            exe.run(fluid.default_startup_program())
+            prog = fluid.default_main_program()
+
+            paddle.save(prog.state_dict(), "./temp-")
+            state_dict_load = paddle.load("./temp-")
+            prog.set_state_dict(state_dict_load)
+        """
+
+        if not isinstance(state_dict, dict):
+            raise TypeError(
+                "Type of `state_dict` should be dict, but received {}.".format(
+                    type(state_dict)))
+
+        vars_dict = {var.name: var for var in self.list_vars()}
```

Comment:
can it compatible with dygraph state_dict?

Reply:
Done,thx.